### PR TITLE
Feat/adding cookies banner

### PR DESCRIPTION
**Pull request by transcendence12**
- create cookies banner component,
- add cookies banner component to layout.tsx

### CODE REVIEW

File: apps/platform-app/src/app/[locale]/(profile)/(components)/CookieBanner/CookieBanner.module.scss [hunk 1]
```diff
@@ -0,0 +1,68 @@
+.bannerWrapper {
+  position: fixed;
+  bottom: 41px;
+  right: 35px;
+  z-index: 1000;
+  display: flex;
+  width: 500px;
+  // height: 322px;
```
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/CookieBanner/CookieBanner.module.scss:8]: **nitpick**: Consider removing the commented-out height property if it's not needed to keep the code clean.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/CookieBanner/CookieBanner.module.scss [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  align-items: flex-start;
+  gap: 22px;
+  flex-shrink: 0;
+
+  border-radius: 9px;
+  border: 1px solid #d9dbe9;
+  background: rgba(255, 255, 255, 0.96);
+  box-shadow: 0px 5px 14px 0px rgba(8, 15, 52, 0.04);
+
+  @include mobile {
```
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/CookieBanner/CookieBanner.module.scss:20]: **suggestion**: Consider using a mixin or variable for the z-index value to maintain consistency across the application.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/CookieBanner/CookieBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { Button } from '@gdh/ui-system'
+import { useEffect, useState } from 'react'
+import styles from './CookieBanner.module.scss'
+
+export const CookieBanner: React.FC = () => {
+  const [isBannerVisible, setIsBannerVisible] = useState<boolean>(false)
+
+  useEffect(() => {
+    try {
+      const hasAcceptedCookies = localStorage.getItem('cookiesAccepted')
```
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/CookieBanner/CookieBanner.tsx:11]: **issue**: Consider adding a check for localStorage availability to enhance compatibility with environments where localStorage is not available.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/CookieBanner/CookieBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    } catch (error) {
+      console.error('Could not set cookiesAccepted in localStorage:', error)
+    }
+  }
+
+  const declineCookies = () => {
+    setIsBannerVisible(false)
+  }
+
+  if (!isBannerVisible) return null
```
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/CookieBanner/CookieBanner.tsx:33]: **suggestion**: Consider using a more descriptive variable name for `isBannerVisible` to clarify its purpose, such as `isCookieBannerVisible`.

**Review by NerdbordMentor — APPROVED** [review]
## Overall Feedback

The implementation of the CookieBanner component meets the task requirements by providing a banner that allows users to accept cookies before using the page. The component is integrated into the layout and styled according to the provided Figma design. The use of localStorage to track cookie acceptance is a practical approach, and the error handling for localStorage access is a good practice.

### Things Missing:

- [] Ensure that the design matches the Figma design exactly, including any specific styling or layout details that may not be visible in the code provided.

### Things That Went Well:

- The CookieBanner component is well-structured and follows React best practices.
- The use of localStorage for managing cookie acceptance is efficient and straightforward.
- The styling is responsive and adapts well to different screen sizes.

**Review by Sumick — CHANGES_REQUESTED** [review]
Code looks great! However we need to implement few more things. Sorry that I haven't specified them before but I've just read about the proper approach in terms of cookie rejection.

1. The scenario when user rejects cookies. Then we should not track him using Google Analytics. I mean, GA tag should not be rendered.
2. If we want to conditionally render GA tag, we will need to have that information server side - so using localStorage as storage for cookie consent selection will not work, as it only works client side. Cookie banner could be server component in our implementation.

**Comment by transcendence12** [comment]
GA4 should collect traces of the user in consent mode. If the user accepts cookies, the information will be stored in the cookie. If the user does not consent ga4 will be active, but the user will not be associated with any identifier. In the developer tools under the network tab you should see `collect`

- create a Google Analytics component,
- add the GA component to the layout, 
- in the cookie banner, the consent information is stored in cookies (values: `granted` or `denied`),
- add privacy policy in the link form banner cookies,
- remove import from next third party google analytics (doesn't offer Consent Mode) and replace by custom component

**Review by Sumick — APPROVED** [review]
Amazing job!!! Congrats ❤️‍🔥 Let's just ensure if we need to remove google analytics package before merging and we are ready to go :)

**Review by filwas — APPROVED** [review]
Lookin' good!

**Pull request merged**